addons/l10n_es_edi_sii/models/account_edi_format.py

```
@@ -0,0 +1,587 @@
+
+            header = {
+                'IDVersionSii': '1.1',
+                'Titular': {
+                    'NombreRazon': invoices.company_id.name[:120],
+                    'NIF': invoices.company_id.vat[2:],
+                },
+                'TipoComunicacion': 'A1' if csv_number else 'A0',
+            }
+
+            # Set registration date
+            invoices.filtered(lambda inv: not inv.l10n_es_registration_date).write({
+                'l10n_es_registration_date': fields.Date.context_today(self),
+            })
+
+            session = requests.Session()
+            session.cert = (cert_file.name, pem_file.name)
+            transport = ESTransport(operation_timeout=60, timeout=60, session=session)
+            client = zeep.Client(connection_vals['url'], transport=transport)
+
+            if invoices[0].is_sale_document():
+                service_name = 'SuministroFactEmitidas'
+            else:
+                service_name = 'SuministroFactRecibidas'
+            if company.l10n_es_edi_test_env and not connection_vals.get('test_url'):
+                service_name += 'Pruebas'
+
+            # Establish the connection.
+            serv = client.bind('siiService', service_name)
+            if company.l10n_es_edi_test_env and connection_vals.get('test_url'):
+                serv._binding_options['address'] = connection_vals['test_url']
+
+            try:
+                if invoices[0].is_sale_document():
+                    res = serv.SuministroLRFacturasEmitidas(header, info_list)
+                else:
+                    res = serv.SuministroLRFacturasRecibidas(header, info_list)
+            except Exception as error:
+                if isinstance(error, requests.exceptions.SSLError):
+                    msg = _("The SSL certificate could not be validated.")
+                elif isinstance(error, zeep.exceptions.Error):
+                    msg = _("Networking error:\n%s") % error
+                else:
```

```suggestion
            except requests.exceptions.SSLError:
                ...
            except zeep.exceptions.Error:
                ...
            except Exception:
               ....
```
also, give a bit more context in your error messages

---

@mart-e but in any of the cases with an error we need to return the error...

---

We will find something...